t_error(3)                                                   t_error(3)

  tt__eerrrroorr(())

  NNAAMMEE

    t_error - produce error message

  SSYYNNOOPPSSIISS

    #include <xti.h>

    int t_error(
        const char *errmsg)

  DDEESSCCRRIIPPTTIIOONN

    The tt__eerrrroorr(3) function produces a message on the standard error output
    which describes the last error encountered during a call to a transport
    function. The argument string errmsg is a user-supplied error message that
    gives context to the error.

    PPaarraammeetteerrss     BBeeffoorree ccaallll     AAfftteerr ccaallll

    errmsg         x               =

    The error message is written as follows: first (if errmsg is not a null
    pointer and the character pointed to be errmsg is not the null character)
    the string pointed to by errmsg followed by a colon and a space; then a
    standard error message string for the current error defined in t_errno. If
    t_errno has a value different from [TSYSERR], the standard error message
    string is followed by a newline character. If, however, t_errno is equal
    to [TSYSERR], the t_errno string is followed by the standard error message
    string for the current error defined in errno followed by a newline.

    The language for error message strings written by tt__eerrrroorr(3) is that of
    the current locale. If it is English, the error message string describing
    the value in t_errno may be derived from the comments following the
    t_errno codes defined in xti.h. The contents of the error message strings
    describing the value in errno are the same as those returned by the
    ssttrreerrrroorr function with an argument of errno.

    The error number, t_errno, is only set when an error occurs and it is not
    cleared on successful calls.

  EEXXAAMMPPLLEE

    If a _tt____cc_oo_nn_nn_ee_cc_tt(3) function fails on transport endpoint fd2 because a bad
    address was given, the following call might follow the failure:

    t_error("t_connect failed on fd2");

    The diagnostic message to be printed would look like:

    t_connect failed on fd2: incorrect addr format

    where incorrect addr format identifies the specific error that occurred,
    and t_connect failed on fd2 tells the user which function failed on which
    transport endpoint.

  VVAALLIIDD SSTTAATTEESS

    All - apart from T_UNINIT

  EERRRROORRSS

    No errors are defined for the tt__eerrrroorr(3) function.

  RREETTUURRNN VVAALLUUEE

    Upon completion, a value of 0 is returned.

  UUSSAAGGEE NNOOTTEESS

    The tt__eerrrroorr function is not thread safe.

    The tt__eerrrroorr function is not async-signal safe.

